ebi_eva_common_pyutils/config_utils.py

```diff
@@ -27,14 +29,58 @@ def __init__(self, settings_xml_file: str):
         with open(settings_xml_file) as xml_file_handle:
             self.config_data = et.parse(xml_file_handle)
 
-    def get_value_with_xpath(self, location: str):
+    def get_value_with_xpath(self, location: str, optional: bool = False):
         etree = self.config_data.getroot()
         result = etree.xpath(location)
-        if not result:
+        if not result and not optional:
             raise ValueError("Invalid XPath location: " + location)
         return result
 
 
+def get_metadata_creds_for_profile(profile_name: str, settings_xml_file: str):
+    """
+    Gets host, username, and password for metadata postgres database.
+    Useful for filling properties files, for connection purposes you can use
+    `metadata_utils.get_metadata_connection_handle`.
+    """
+    properties = get_properties_from_xml_file(profile_name, settings_xml_file)
+    pg_url = properties['eva.evapro.jdbc.url']
+    pg_user = properties['eva.evapro.user']
+    pg_pass = properties['eva.evapro.password']
+    return pg_url, pg_user, pg_pass
+
+
+def get_primary_mongo_creds_for_profile(profile_name: str, settings_xml_file: str):
+    """
+    Gets primary host, username, and password for mongo database.
+    Useful for filling properties files, for connection purposes it is preferable to use
+    `mongo_utils.get_mongo_connection_handle` as that will handle multiple hosts appropriately.
+    """
+    properties = get_properties_from_xml_file(profile_name, settings_xml_file)
```

Comment:
We might need an optional argument to force the use of the first or another function that return the secondary.